File: common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/injection/GeneratorComponent.java
package io.vrap.rmf.codegen.common.generator.injection;

import com.google.inject.Guice;
import com.google.inject.Injector;
import io.vrap.rmf.codegen.common.generator.core.STCodeGenerator;


public final class GeneratorComponent {

    private final Injector injector ;

    public GeneratorComponent(GeneratorModule generatorModule) {
        this.injector = Guice.createInjector(generatorModule);
    }

    public STCodeGenerator getStCodeGenerator(){
        return injector.getInstance(STCodeGenerator.class);
    }

}

File: common-codegen-kt/src/main/kotlin/io/vrap/rmf/codegen/kt/di/GeneratorComponent.kt
package io.vrap.rmf.codegen.kt.di

import com.google.inject.Guice
import com.google.inject.Module
import io.vrap.rmf.codegen.kt.rendring.CoreCodeGenerator


class GeneratorComponent(generatorModule: GeneratorModule, vararg modules:  Module) {

    private val injector = Guice.createInjector(listOf(generatorModule).plus(modules))

    private val coreCodeGenerator = injector.getInstance(CoreCodeGenerator::class.java)

    fun generateFiles() = coreCodeGenerator.generate()

}